del(6x legacy forms): delete

NathalieC opened the pull request:
* delete page
* take note to add the warning on the design forms doc pages

NathalieC committed:
del(6x legacy forms): delete

* delete page
* take note to add the warning on the design forms doc pages

NathalieC committed:
feat(bonita-bpm-overview): add warning 6.x forms

* add warning about 6.x forms not working in 7.8
* remove parts when it says 6.x will continue to work

educhastenier commented:
Pages:
* page-and-form-development-overview.md
* taxonomy.md
* comparison-of-7-x-and-6-x.md

still references the deleted page. Please remove those references.

danila-m committed:
Remove references to 6.x
Remove references to 5.x

Rewrite some of the pages to be more coherent
Fix dead links.
Add a step-by-step guide to 7.8.0 migration.

NathalieC commented on md/bonita-bpm-overview.md:
```diff
@@ -58,7 +61,7 @@ See [Lifecycle and profiles](lifecycle-and-profiles.md).
 **Can I migrate an existing process into Bonita 7.x?**
 
-You can [import a process](import-and-export-a-process.md) from any earlier Bonita release. You can also [migrate a process from Bonita Open Solution 5.9 or 5.10](migrate-a-process-from-bonita-open-solution-5-x.md) to this release. A 6.x process will continue to run unchanged in 7.x, but to take advantage of the new features, you will need to update the process and [migrate the forms](migrate-a-f ...
+You can [import a process](import-and-export-a-process.md) from any earlier Bonita release. Since 6.x forms don't work in Bonita 7.8, you may want to read how to [migrate a form from 6.x](migrate-a-form-from-6-x.md).
```
You can [import a process](import-and-export-a-process.md) from any earlier Bonita release if  they don't contain Legacy v6 forms or case overview pages. If they do, you need to rework your process and [migrate those froms or pages](migrate-a-form-from-6-x.md).

NathalieC commented on md/comparison-of-7-x-and-6-x.md:
```diff
@@ -18,27 +18,27 @@ See the [Installation overview](bonita-bpm-installation-overview.md).
 | Environments  | An Environment is a set of configuration information for a particular process. It contains the following information:  actor mapping, actor filters, parameters, connector implementations, dependencies for processes and applications. By having these Environments set up in the Studio, the developer can export processes that are ready to be run for a specific target platform. At an ...
 | Parameters  | A Parameter is configuration information that can be attached to a process. Parameters are configured for an Environment. Parameter values can be updated at runtime. | Now available in all editions. See [Parameters](parameters.md) and [Environments](environments.md)  |
-| Workspace API  | The Workspace API is provided to generate `.bar` files from the command line, ready for production. .BAR files are for deployment only and cannot be read by the Studio. .BOS files are for Studio interchange only and cannot be deployed to production. Each file format is optimal in term of content and is optimized for the target use case  | No change. The Workspace API can export ...
+| Workspace API  | The Workspace API is provided to generate `.bar` files from the command line, ready for production. .BAR files are for deployment only and cannot be read by the Studio. .BOS files are for Studio interchange only and cannot be deployed to production. Each file format is optimal in term of content and is optimized for the target use case  | The Workspace API behaviour does not ch ...
```
```suggestion
| Workspace API  | The Workspace API is provided to generate `.bar` files from the command line, ready for production. .BAR files are for deployment only and cannot be read by the studio. .bos files are for Studio interchange only and cannot be deployed to production. Each file format is optimal in term of content and is optimized for the target use case  | The Workspace API can export forms that are mapped to process instantiation or tasks, and pages that are mapped to case overview. It cannot export application pages that are not elements of a process. The Workspace API has been deprecated in 7.7.0, and is replaced by the *LA builder* included in the tooling suite of [*Bonita Continuous Delivery* add-on](https://documentation.bonitasoft.com/bcd/2.0/). See [Automating builds](automating-builds.md)  |
```

NathalieC commented on md/comparison-of-7-x-and-6-x.md:
```diff
@@ -18,27 +18,27 @@ See the [Installation overview](bonita-bpm-installation-overview.md).
 | -------- | --------------- | --------------- |
-| UI designer  | Not in 6.x  | A new development environment for creating sophisticated, data-driven pages and forms for process-based applications. The UI designer is accessed through the Studio. See [live update](live-update.md). The 6.x [form tooling](6-x-legacy-forms.md) is still supported, so that you can continue to run 6.x processes and to ease migration  |
-| Process migration  | A process exported from Bonita Open Solution 5.9 or 5.10 must be modified to run in Bonita. When you import a process from 5.9 or 5.10 into Bonita Studio, you see a report that guides you through the changes that are necessary. See [Migrate a process from Bonita Open Solution 5.9 or 5.10](migrate-a-process-from-bonita-open-solution-5-x.md)  | A process created in Bonita 6.x ...
+| UI designer  | Not in 6.x  | A new development environment for creating sophisticated, data-driven pages and forms for process-based applications. The UI designer is accessed through the Studio. See [live update](live-update.md). The 6.x forms are supported up to the 7.7.4, so that you can continue to run 6.x processes and to ease migration  |
```
```suggestion
| UI Designer  | Not in 6.x  | A new development environment for creating sophisticated, data-driven pages and forms for process-based applications. The UI Designer is accessed through the studio. See [live update](live-update.md). The 6.x forms were supported up to the 7.7.x, so you could continue to run 6.x processes and ease migration.  |
```

NathalieC commented on md/comparison-of-7-x-and-6-x.md:
```diff
@@ -18,27 +18,27 @@ See the [Installation overview](bonita-bpm-installation-overview.md).
-| UI designer  | Not in 6.x  | A new development environment for creating sophisticated, data-driven pages and forms for process-based applications. The UI designer is accessed through the Studio. See [live update](live-update.md). The 6.x [form tooling](6-x-legacy-forms.md) is still supported, so that you can continue to run 6.x processes and to ease migration  |
-| Process migration  | A process exported from Bonita Open Solution 5.9 or 5.10 must be modified to run in Bonita. When you import a process from 5.9 or 5.10 into Bonita Studio, you see a report that guides you through the changes that are necessary. See [Migrate a process from Bonita Open Solution 5.9 or 5.10](migrate-a-process-from-bonita-open-solution-5-x.md)  | A process created in Bonita 6.x ...
+| UI designer  | Not in 6.x  | A new development environment for creating sophisticated, data-driven pages and forms for process-based applications. The UI designer is accessed through the Studio. See [live update](live-update.md). The 6.x forms are supported up to the 7.7.4, so that you can continue to run 6.x processes and to ease migration  |
+| Process migration  | N/A| A process created in Bonita 6.x will run in 7.x (up to 7.7.4) without any modification. However, to take advantage of the new features in Bonita 7.x, you need to add contracts and migrate the forms. See [Contracts](contracts-and-contexts.md) and [Migrate a form from 6.x](migrate-a-form-from-6-x.md)  |
```
```suggestion
| Process migration  | N/A| A process created in Bonita 6.x could run in 7.x (up to 7.7.x) without any modification. Starting with Bonita 7.8, you must use contracts create froms outside of the studio, or migrate the forms. See [Contracts](contracts-and-contexts.md) and [Migrate a form from 6.x](migrate-a-form-from-6-x.md)  |
```

NathalieC commented on md/comparison-of-7-x-and-6-x.md:
```diff
@@ -18,27 +18,27 @@ See the [Installation overview](bonita-bpm-installation-overview.md).
 | Form mapping  | Not in 6.x  | A form is mapped to process instantiation, to case overview, or to a task. Any form that respects the relevant contract can be used. This separation between the process and the forms makes it possible to to the live update feature  |
 | Business data  | A process or application can use business data managed by the Bonita Platform. (Subscription editions only)  | Now available in all editions. See [BDM](define-and-deploy-the-bdm.md)  |
-| Actor mapping  | Each actor in a process can be mapped to a Group, Role, Membership or User within an organization. The mapping can be refined using an actor filter.  | No change. See [Actor mapping](actors.md) and [Actor filtering](actor-filtering.md)  |
+| Actor mapping  | Each actor in a process can be mapped to a Group, Role, Membership or User within an organization. The mapping can be refined using an actor filter.  | No change. Anonymous users are not supported anymore. See [Actor mapping](actors.md) and [Actor filtering](actor-filtering.md)  |
```
You state the absence of anonymous user further down the table. Maybe it's not needed here.
Not very proud we don't support it anymore :/

danila-m replied:
We don't support it from 7.2, and the "support" since 7.0 was more of a hack than a feature.
The use of anonymous users was seen as bad practice at this point, hence why no effort was put to maintain it. Anyway right now it is history/a new PF as far as the product goes, and considering the purpose of this doc page, given it's one of the few point we don't support from 6.x, I feel repeating it on a related subject is not too bad.

NathalieC replied:
Fine by me.

NathalieC commented on md/comparison-of-7-x-and-6-x.md:
```diff
@@ -47,11 +47,11 @@ See the [Installation overview](bonita-bpm-installation-overview.md).
 | Task management  | Users can choose how to manage tasks. They can perform one task after another in list order, or select my tasks for themselves then perform them in the order they choose.  | No change for users. An administrator or process manager (or a user with an appropriate custom profile) can see at a glance that status of all tasks in a case, and can perform a task for a user. See [Mana ...
 | Mobile web portal  | A specially designed mobile interface to Bonita Portal enables users to carry out tasks from a browser on mobile devices  | No change. See [Mobile overview](mobile-portal.md)  |
-| Subtasks  | A Subtask is a part of a self-assigned existing task. A Subtask must be assigned to a specific person, by name. The assignee can be the creator  | No change. See [Manage a subtask](subtasks.md)  |
+| Subtasks  | A Subtask is a part of a self-assigned existing task. A Subtask must be assigned to a specific person, by name. The assignee can be the creator  | Supported up to 7.2.4. No more substasks can be created starting from 7.3.0, but ongoing process instances should execute themselves correctly. See [Manage a subtask](subtasks.md)  |
```
```suggestion
| Subtasks  | A subtask is a part of an existing task. The assignee can create a subtask (to request a missing information for example) and must assign it to a specific person, by name. The assignee can be the creator  | Supported up to 7.2.4. No more substasks can be created starting from 7.3.0, but open process instances execute them correctly. See [Manage a subtask](subtasks.md)  |
```

NathalieC commented on md/comparison-of-7-x-and-6-x.md:
```diff
@@ -47,11 +47,11 @@ See the [Installation overview](bonita-bpm-installation-overview.md).
 | Feature  | Bonita 6.x  | Bonita 7.0/1/2  | Bonita 7.3 | 
 | -------- | --------------- | ------------------- | -------------- | 
-| bonita_home  | A separate bonita_home for each edition and for cluster  | A common  bonita_home used for all editions, simplifying download and installation. Improved structure for bonita_home/server that is easier to maintain and customize. See [Bonita Home](bonita-home.md)  | No more bonita_home. See  [platform setup](BonitaBPM_platform_setup.md) 
+| bonita_home  | A separate bonita_home for each edition and for cluster  | A common  bonita_home used for all editions, simplifying download and installation. Improved structure for bonita_home/server that is easier to maintain and customize.  | No more bonita_home. All the relevant info is store in the database, increasing performance and easing future migrations. See  [platform setup](BonitaBP ...
```
```suggestion
| bonita_home  | A separate bonita_home for each edition and for cluster  | A common  bonita_home used for all editions, simplifying download and installation. Improved structure for bonita_home/server that is easier to maintain and customize.  | No more bonita_home. Relevant information is stored in the database, increasing performance and easing future migrations. See  [platform setup](BonitaBPM_platform_setup.md) 
```

NathalieC commented on md/migrate-a-form-from-6-x.md:
```diff
@@ -1,12 +1,9 @@
 ::: warning
-**Attention**: By the end of 2018, V6 GWT forms won't be available for modeling or execution.
-We strongly advise you to switch to forms created with Bonita UI Designer to benefit from technologies like html/AngularJS and use contracts in tasks and process instantiation.
+**Attention**: Starting with Bonita 7.8, V6 GWT forms are not available for modeling or execution. To perform the operations listed below, you will need to import your processes in a Bonita Studio of any version up to 7.7.4, and build new versions of your processes before migrating to 7.8.0 or above.
```
```suggestion
**Attention**: Starting with Bonita 7.8, 6.x GWT forms and case overview forms are not available for both modeling and execution. To perform the operations listed below, you will need to import your processes in a Bonita Studio of any version up to 7.7.x, and build new versions of your processes before migrating to 7.8.0 or above.
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
 
+## Migration of processes with v6 forms
+
+Until the version 7.0.0, Bonita used forms based on the GWT technology, for the instantiation of processes, tasks and process overview.
```
```suggestion
Until the version 7.0.0, Bonita used UI artifacts based on the Google Web Toolkit (GWT) technology: process instantiation and task execution forms, and case overview page.
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+## Migration of processes with v6 forms
+
+Until the version 7.0.0, Bonita used forms based on the GWT technology, for the instantiation of processes, tasks and process overview.
+The runtime support for those forms was removed in 7.8.0.
```
```suggestion
The runtime support for those forms and page was removed in 7.8.0.
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+Until the version 7.0.0, Bonita used forms based on the GWT technology, for the instantiation of processes, tasks and process overview.
+The runtime support for those forms was removed in 7.8.0.
+
+It means, that if one or more processes on the migrated server uses v6 forms, the migration to a version above 7.7.4 cannot be performed directly. The following lines explain how to migrate to a version 7.8.0 +
```
```suggestion
It means that if one or more processes on the migrated server uses 6.x forms or overview page, the migration to a version above 7.7.x cannot be performed directly. The following lines explain how to migrate to a version 7.8.0 +
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+
+Specifically if you are migrating from a 6.x version:
+* Migrate to the 7.0.0 using the migration tool 1.x.
+* Migrate to the 7.7.4 using the migration tool 2.x.
```
```suggestion
* Migrate to the last 7.7.x version, using the migration tool 2.x.
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+Specifically if you are migrating from a 6.x version:
+* Migrate to the 7.0.0 using the migration tool 1.x.
+* Migrate to the 7.7.4 using the migration tool 2.x.
+* Redesign all your forms in the Studio using the UI designer. See [here](migrate-a-form-from-6-x.md) for more info.
```
```suggestion
* Redesign your process to use contracts at process instantiation and task execution levels, and recreate all your forms and case ovierview pages in the Studio using the UI Designer or your favorite IDE, so they use the contracts. For more information, go to [migrate a form from 6.x](migrate-a-form-from-6-x.md).
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+* Migrate to the 7.0.0 using the migration tool 1.x.
+* Migrate to the 7.7.4 using the migration tool 2.x.
+* Redesign all your forms in the Studio using the UI designer. See [here](migrate-a-form-from-6-x.md) for more info.
+* Upload the new version of all your processes using the new forms.
```
```suggestion
* Upload the new version of all your processes using the contracts, new forms, and new case overview pages.
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+* Migrate to the 7.7.4 using the migration tool 2.x.
+* Redesign all your forms in the Studio using the UI designer. See [here](migrate-a-form-from-6-x.md) for more info.
+* Upload the new version of all your processes using the new forms.
+* Disable the version of your processes using v6 forms. Make sure they have no more running instances.
```
```suggestion
* Make sure the versions of the processes using 6.x forms have no more open instances, and disable them.
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+* Disable the version of your processes using v6 forms. Make sure they have no more running instances.
+* Perform the migration to the desired version.
+
+If you are migrating from a 7.x version :
```
```suggestion
If you are migrating from a 7.x version:
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+* Perform the migration to the desired version.
+
+If you are migrating from a 7.x version :
+* Redesign all your forms in the Studio using the UI designer. See [here](migrate-a-form-from-6-x.md) for more info.
```
What is the difference between 6.x and 7.x with 6.x forms?

danila-m replied:
None, aside of the fact that you can't redesign your 6.x forms in a 6.5.4 ;).

NathalieC replied:
Alright. The sentences look the same, so I was wondering why not making them only one sentence. Not a big deal anyway.

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+
+The disabled processes with v6 forms will not be able to be enabled again post migration.
+Having v6 overviews on your processes will not prevent the migration of the platform, however they will all be replaced by the default v7 overview in the process.
+It means that you might want redo the overview of your processes as well as the forms, especially if you have configured a custom overview for your processes in v6.
```
or (for Enterprise, Performance, and Efficiency only) live update it after migration.

NathalieC replied:
```suggestion
It means that you might want to redo the case overview page as well as the forms, especially if you have configured a custom case overview page for your processes in version 6.x.
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+* Perform the migration to the desired version.
+
+The disabled processes with v6 forms will not be able to be enabled again post migration.
+Having v6 overviews on your processes will not prevent the migration of the platform, however they will all be replaced by the default v7 overview in the process.
```
```suggestion
Having 6.x case overview pages on your processes will not prevent the migration of the platform, however they will all be replaced by the default 7.x case overview page, created with the UI Designer, in the processes.
```

NathalieC commented on md/migration-overview.md:
```diff
@@ -15,14 +15,11 @@ your workspace is not automatically migrated. In this case, you must manually ex
-You can import a .bos file containing a process definiiton from a Bonita 6.x or Bonita Open Solution 5.x. 
-The [guided migration](migrate-a-process-from-bonita-open-solution-5-x.md) converts the 5.x process definition to use the legacy 6.x tooling so that the process can run on a Bonita 7.x platform. 
-You can modify the process definition later to take advantage of the new features in this version of Bonita.
+A process definition created using a version 6.x of Bonita, or using 6.x style forms or overviews, needs to have its task and instantiation forms modified in a previous version of 
```
```suggestion
A process definition created using a version 6.x of Bonita, or using 6.x forms or case overview pages, needs to have its task and instantiation forms modified in a previous version of 
```

NathalieC commented on md/page-and-form-development-overview.md:
```diff
@@ -13,11 +13,9 @@ You can also use external forms referenced by URL. If an external URL is being u
 ::: warning
-**Attention**: By the end of 2018, V6 GWT forms won't be available for modeling or execution.
-We strongly advise you to switch to forms created with Bonita UI Designer to benefit from technologies like html/AngularJS and use contracts in tasks and process instantiation.
+**Attention**: The 7.8 version of Bonita doesn't support V6 GWT forms either at design or at runtime.
```
```suggestion
**Attention**: Bonita 7.8 doesn't support 6.x GWT forms and case overview pages, neither for design nor at runtime.
```

NathalieC commented on md/bonita-bpm-overview.md:
```diff
@@ -45,6 +45,9 @@ Bonita is provided in five different editions: Community, Teamwork, Efficiency,
 
 Bonita 7.x provides new and improved features, including the [UI Designer](ui-designer-overview.md) for creating application pages and forms, as well as [contracts and context](contracts-and-contexts.md) to create a clear split between the process logic, the data, and the user views.   
+
+::: warning Attention: Starting with Bonita 7.8, V6 GWT forms are not available for modeling or execution. Before migrating to Bonita 7.8, you need to create all forms with Bonita UI Designer to benefit from technologies like html/AngularJS and use contracts in tasks and process instantiation. For more information, read the [migrate a form from 6.x](migrate-a-form-from-6-x.md) page. :::
```
```suggestion
::: warning Attention: Starting with Bonita 7.8, 6.x forms and case overview pages are not available for modeling or execution. Before migrating to Bonita 7.8, you need to create all forms with Bonita UI Designer to benefit from technologies like html/AngularJS and use contracts in tasks and process instantiation. For more information, read the [migrate a form from 6.x](migrate-a-form-from-6-x.md) page. :::
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
 
 The migration is now complete. If you were using a custom Look & Feel before migration, test it on the new version before applying it to your migrated platform.
 
+## Migration of processes with v6 forms
```
```suggestion
## Migration of processes with 6.x forms and case overview pages
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+* Disable the version of your processes using v6 forms. Make sure they have no more running instances.
+* Perform the migration to the desired version.
+
+The disabled processes with v6 forms will not be able to be enabled again post migration.
```
```suggestion
The disabled processes with 6.x forms will not be able to be enabled again post migration.
```

NathalieC commented on md/migrate-from-an-earlier-version-of-bonita-bpm.md:
```diff
@@ -222,6 +222,31 @@ If you did not set the default Look & Feel before migration and you cannot log i
+If you are migrating from a 7.x version :
+* Redesign all your forms in the Studio using the UI designer. See [here](migrate-a-form-from-6-x.md) for more info.
+* Upload the new version of all your processes using the new forms.
+* Disable the version of your processes using v6 forms. Make sure they have no more running instances.
```
```suggestion
* Disable the version of your processes using 6.x forms. Make sure they have no more running instances.
```

NathalieC reviewed (commented):
review done

danila-m committed:
feedback integration

NathalieC commented:
I don't see in gitHub how to solve the conflict. Can you help here?
